ibis/backends/tests/test_array.py

```diff
@@ -24,3 +25,14 @@ def test_array_concat(backend, con):
 def test_array_length(backend, con):
     expr = ibis.literal([1, 2, 3]).length()
     assert con.execute(expr) == 3
+
+
+@pytest.mark.xfail_unsupported
+@pytest.mark.skip_missing_feature(
+    ['supports_arrays', 'supports_arrays_outside_of_select']
+)
+def test_np_array(backend, con):
+    arr = np.array([1, 2, 3])
```

Comment:
can you also try with `np.array(1)` e.g. a 0-dim scalar. migth as well add for 2-d too.

Reply:
I tried this out for the Pandas and PySpark backends, and currently, support for 0-d or 2-d (or higher) arrays isn't great. I'm not sure whether it's worth supporting them (at least for now) because arrays being 1-d seems pretty baked into how array ops are implemented in Ibis.

0-d arrays we could maybe convert to scalars in Ibis. If it's not a big use case though I am thinking that simply supporting only 1-d numpy arrays would be the best way to go

Reply:
(in that case this PR is more just a slight extension on top of Ibis accepting Python lists for arrays (now also accepts 1d numpy arrays, which are list-like), rather than truly supporting ndarrays in their full form)